### PR TITLE
split and rename insight into smaller ones

### PR DESCRIPTION
**Pull request by Stefan-Stojanovic**

### CODE REVIEW

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/python/functional-programming/immutability/more-benefits-of-immutability.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/python/functional-programming/immutability/more-benefits-of-immutability.md?ref=07d5d47bb431e23a912d518f17d30297bd282e6d)

**Comment by enki-hq** [comment]
Rule workout:insights failed for files: 
  - python/functional-programming/immutability/README.md

**Comment by enki-hq** [comment]
Rule workout:insights failed for files: 
  - python/functional-programming/immutability/README.md

File: python/functional-programming/immutability/atomicity-of-failure.md [hunk 1]
```diff
@@ -1,141 +1,122 @@
...
+b3 = MutableShoppingBasket(b2.itemcount-1)
+print(b3)
+# Traceback (most recent call last):
+#   File "python", line 27, in <module>
+#   File "python", line 4, in __init__
+# ValueError: You can't have less than
+#    zero items in the basket!
+```
+
+It doesn't let us, as, in creating the new object, we fail to satisfy the constraints when the constructor is called. The object is not created, and it is easy to spot where we've went wrong due to failure being contained at this point and this point only. (*As opposed to silently creating a broken object to cause trouble elsewhere.*) This is another great advantage of immutable objects as it provides this without much extra effort on the developer's part.
```
Comment (kapnobatai137) [inline python/functional-programming/immutability/atomicity-of-failure.md:85]:
```suggestion
It won't work. When creating the new object, we fail to satisfy the constraints when the constructor is called. The object is not created, and it is easy to spot where we've went wrong due to failure being contained at this point and this point only (*as opposed to silently creating a broken object to cause trouble elsewhere*). This is another great advantage of immutable objects as it provides this without much extra effort on the developer's part.
```

**Pull request merged**